### ISSUE TITLE
False Positive: link-in-text-block on "Label: <link>" contact/credit lines

### ISSUE THREAD

**Issue by chutchins25** [dequelabs/axe-core#5220]
**Product:** axe-core
**Product Version:** 4.12.1
**Latest Version:** I have tested the issue with the latest version of the product

#### Expectation

`link-in-text-block` should flag links that a reader could lose inside a block of running text. A short `Label: <link>` line (e.g. a contact or credit line) is not running prose, so it should not be reported.

#### Actual

`link-in-text-block` reports a `serious` violation on a link that is the value in a short `Label: <link>` line, even though it is not embedded in a sentence:

```
The link has insufficient color contrast of 2.72:1 with the surrounding text.
(Minimum contrast is 3:1, link text: #ffffff, surrounding text: #9b9d9f)
The link has no styling (such as underline) to distinguish it from the surrounding text
```

The contrast math is correct: the container text color is `rgba(255,255,255,0.55)`, which flattens over the `#212529` background to `#9b9d9f`, while the link is full white `#ffffff` with `text-decoration: none`. So `contrast(#ffffff, #9b9d9f) = 2.72:1`, just under the 3:1 threshold, and there is no non-color distinction. The link is nonetheless flagged despite being a single `Label: name` line rather than a link inside prose.

#### How to Reproduce

Minimal, self-contained repro:

```html
<body style="background:#212529;">
  <div style="color: rgba(255,255,255,0.55);">
    Written by: <a href="mailto:jane@example.com" style="color:#ffffff; text-decoration:none;">Jane Doe</a>
  </div>
</body>
```

Run `axe.run(document, { runOnly: ['link-in-text-block'] })` → one `serious` violation on the `<a>` (contrast `2.72:1`, colors `#ffffff` / `#9b9d9f`).

#### Additional context

Likely (unconfirmed) root cause: the `isInTextBlock` matcher treats these lines as text blocks because its length heuristic — `parentText.length > widgetText.length` — trips whenever the label text is longer than the link text (`"Written by:"` = 11 vs `"Jane Doe"` = 8). A short label followed by a name isn't the "link buried in prose" case the rule targets. Flagged as a hypothesis; leaving the fix to maintainers.

Not a duplicate of the existing matcher issues #3841 (inline-block), #4126 (parent nodes/images), #4255 (`<br>`), #4602 (underline methods), or #4167 (comparison node).

**Comment by WilcoFiers** [comment]
Ooh! This isn't a false positive at all. That link text looked much bolder to me than the other text, but it isn't. It's purely a color change, and not one with enough contrast. It has 2.7:1, but should be 3:1.
